server/libs/infra/src/repositories/partner.repository.ts

```diff
@@ -0,0 +1,62 @@
+import { IPartnerRepository, PartnerDirection } from '@app/domain';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PartnerEntity, UserEntity } from '../entities';
+import { CreatePartnerDto } from '@app/domain/partner/dto';
+
+@Injectable()
+export class PartnerRepository implements IPartnerRepository {
+  readonly logger = new Logger(PartnerRepository.name);
+
+  constructor(
+    @InjectRepository(PartnerEntity)
+    private readonly repository: Repository<PartnerEntity>,
+  ) {}
+
+  getAll(userId: string, direction: PartnerDirection): Promise<PartnerEntity[]> {
+    return this.repository.find({
+      where: {
+        ...(direction === 'shared-with' && { sharedWith: { id: userId } }),
+        ...(direction === 'shared-by' && { sharedBy: { id: userId } }),
+      },
+      relations: {
+        sharedBy: true,
+        sharedWith: true,
+      },
+    });
+  }
+
+  async get(sharedBy: string, sharedWith: string): Promise<PartnerEntity | null> {
+    return this.repository.findOne({
+      where: {
+        sharedWith: {
+          id: sharedWith,
+        },
+        sharedBy: {
+          id: sharedBy,
+        },
+      },
+      relations: {
+        sharedBy: true,
+        sharedWith: true,
+      },
+    });
+  }
+
+  create(createPartnerDto: CreatePartnerDto): Promise<PartnerEntity> {
+    const partner = new PartnerEntity();
+
+    partner.sharedWith = new UserEntity();
+    partner.sharedWith.id = createPartnerDto.sharedWith;
+
+    partner.sharedBy = new UserEntity();
+    partner.sharedBy.id = createPartnerDto.sharedBy;
+
+    return this.repository.save(partner);
```

Comment:
This works fine, but I think you could also do

```typescript
return this.repository.save({ sharedWithId: dto.sharedWithId: sharedById: dto.sharedBy });
```

AFAIK setting `sharedWithId` is the equivalent to setting `sharedWith: { id: sharedWithId }`. At the end of the day TypeORM is just going to write values to those two columns (plus the computed ones).